Sources/PubNub/PubNub.swift

```diff
@@ -32,13 +32,11 @@ public struct PubNub {
   /// Instance identifier
   public let instanceID: UUID
   /// A copy of the configuration object used for this session
-  public let configuration: PubNubConfiguration
+  public var configuration: PubNubConfiguration
```

Comment:
What is the thought of making this a public mutable vs internal/private mutable?  I'm assuming it's solely for the `set(token:)` method below?  

Tokens could be either changed by passing in a custom configuration per-request and/or this could be a private set to allow for the `set(token:)` below if the user doesn't want to manager their own configuration.